From 012ee7c3fe24a5e75d9b0465268c1bb8187b8337 Mon Sep 17 00:00:00 2001
From: Jakub Hrozek <jhrozek@redhat.com>
Date: Wed, 15 Mar 2017 18:00:37 +0100
Subject: [PATCH 01/97] Updating the version for the 1.15.3 release

---
 version.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/version.m4 b/version.m4
index 11e5e71b08b8af7c8d4f39410fea99d5d2ce3a00..da96ecd2e2d58e3e6079741f81a07f99259010b4 100644
--- a/version.m4
+++ b/version.m4
@@ -1,5 +1,5 @@
 # Primary version number
-m4_define([VERSION_NUMBER], [1.15.2])
+m4_define([VERSION_NUMBER], [1.15.3])
 
 # If the PRERELEASE_VERSION_NUMBER is set, we'll append
 # it to the release tag when creating an RPM or SRPM
-- 
2.12.2

