From 574981ba9b1c7e4c0455ea85d951ebc2817ebf54 Mon Sep 17 00:00:00 2001
From: Stephen Gallagher <sgallagh@redhat.com>
Date: Mon, 16 Jul 2012 09:46:24 -0400
Subject: [PATCH] AD: Add missing DP option terminator

---
 src/providers/ad/ad_opts.h |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/providers/ad/ad_opts.h b/src/providers/ad/ad_opts.h
index 2648044c0079ed92b7c24ee7eac54af1d2b61a13..9f950a803330a1972c49e71fb1f99f22cde6fd35 100644
--- a/src/providers/ad/ad_opts.h
+++ b/src/providers/ad/ad_opts.h
@@ -134,7 +134,8 @@ struct dp_option ad_def_krb5_opts[] = {
     { "krb5_renew_interval", DP_OPT_NUMBER, NULL_NUMBER, NULL_NUMBER },
     { "krb5_use_fast", DP_OPT_STRING, NULL_STRING, NULL_STRING },
     { "krb5_fast_principal", DP_OPT_STRING, NULL_STRING, NULL_STRING },
-    { "krb5_canonicalize", DP_OPT_BOOL, BOOL_TRUE, BOOL_TRUE }
+    { "krb5_canonicalize", DP_OPT_BOOL, BOOL_TRUE, BOOL_TRUE },
+    DP_OPTION_TERMINATOR
 };
 
 struct sdap_attr_map ad_2008r2_attr_map[] = {
-- 
1.7.10.4

