From cc1bae136c35b0a72c0e39928b0be414e182d375 Mon Sep 17 00:00:00 2001
From: Jakub Hrozek <jhrozek@redhat.com>
Date: Thu, 10 Nov 2011 12:36:26 +0100
Subject: [PATCH] configAPI: Fix removing in old domain when saving a new
 domain

---
 src/config/SSSDConfig.py |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/config/SSSDConfig.py b/src/config/SSSDConfig.py
index 87b1d63..6858ce4 100644
--- a/src/config/SSSDConfig.py
+++ b/src/config/SSSDConfig.py
@@ -1760,7 +1760,7 @@ class SSSDConfig(SSSDChangeConf):
             domain.oldname = None;
 
         sectionname = 'domain/%s' % name
-        section_subtree = self.findOpts(self.opts, 'section', sectionname)
+        (no, section_subtree) = self.findOpts(self.opts, 'section', sectionname)
 
         if name not in self.list_domains():
             self.add_section(sectionname, []);
@@ -1768,7 +1768,7 @@ class SSSDConfig(SSSDChangeConf):
         for option in self.options(sectionname):
             if option['type'] == 'option':
                 if option['name'] not in domain.get_all_options():
-                    self.delete_option_subtree(section_subtree, 'option', option['name'], True)
+                    self.delete_option_subtree(section_subtree['values'], 'option', option['name'], True)
 
         for option,value in domain.get_all_options().items():
             if (type(value) == list):
-- 
1.7.7

