From 07a71386a04d4b1860a4250ea57fd5866ea792fe Mon Sep 17 00:00:00 2001
From: Lukas Slebodnik <lslebodn@redhat.com>
Date: Mon, 9 Feb 2015 17:40:07 +0100
Subject: [PATCH 01/30] BUILD: Remove unused libraries for pysss.so

Reviewed-by: Stephen Gallagher <sgallagh@redhat.com>
---
 Makefile.am | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index ac6a358ea14239781c26e6f2ac02bdeb3007659f..bdca46a0548ddb98dde485a7786566eb623b181b 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -445,15 +445,9 @@ SSSD_LIBS = \
 
 PYTHON_BINDINGS_LIBS = \
     $(TALLOC_LIBS) \
-    $(TEVENT_LIBS) \
     $(POPT_LIBS) \
     $(LDB_LIBS) \
-    $(DBUS_LIBS) \
-    $(PCRE_LIBS) \
-    $(DHASH_LIBS) \
-    $(SSS_CRYPT_LIBS) \
-    $(OPENLDAP_LIBS) \
-    $(TDB_LIBS)
+    $(NULL)
 
 TOOLS_LIBS = \
     $(LTLIBINTL) \
-- 
2.4.3

