From 6fb643c75e59e093c8c52def162ce1f1956430c7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C5=BDidek?= <mzidek@redhat.com>
Date: Tue, 14 Mar 2017 18:20:43 +0100
Subject: [PATCH 02/97] UTIL: Typo in comment
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Reviewed-by: Lukáš Slebodník <lslebodn@redhat.com>
---
 src/util/safe-format-string.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/util/safe-format-string.h b/src/util/safe-format-string.h
index 2f4796de7cea66d9ff0cd808e9e7c33de053feb8..fdebcf966554bcce150ed531173e991973698da0 100644
--- a/src/util/safe-format-string.h
+++ b/src/util/safe-format-string.h
@@ -42,7 +42,7 @@
  * Features:
  * - Only string 's' fields are supported
  * - All the varargs should be strings, followed by a NULL argument
- * - Both positional '%$1s' and non-positional '%s' are supported
+ * - Both positional '%1$s' and non-positional '%s' are supported
  * - Field widths '%8s' work as expected
  * - Precision '%.8s' works, but precision cannot be read from a field
  * - Left alignment flag is supported '%-8s'.
-- 
2.12.2

