From 0b6df55aee996a4b1e8824d1c58c5494b0c5fb0b Mon Sep 17 00:00:00 2001
From: Ariel Barria <arielb@fedoraproject.org>
Date: Sat, 12 May 2012 11:00:51 -0500
Subject: [PATCH 02/19] Potential NULL dereference in proxy provider

---
 src/providers/proxy/proxy_id.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/providers/proxy/proxy_id.c b/src/providers/proxy/proxy_id.c
index 8a8c7ca80d1b24e53c3d55d06564e719a069642a..e7d9206e5081153ef389dd25db7a32816cc44839 100644
--- a/src/providers/proxy/proxy_id.c
+++ b/src/providers/proxy/proxy_id.c
@@ -215,7 +215,7 @@ static int save_user(struct sysdb_ctx *sysdb, bool lowercase,
         shell = NULL;
     }
 
-    if (!lowercase || alias) {
+    if (lowercase || alias) {
         attrs = sysdb_new_attrs(NULL);
         if (!attrs) {
             DEBUG(SSSDBG_CRIT_FAILURE, ("Allocation error ?!\n"));
-- 
1.7.10.1

