From 8c8f74b0dfa29643279d31b12300ced47d5c2ab5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pavel=20B=C5=99ezina?= <pbrezina@redhat.com>
Date: Thu, 16 Aug 2018 11:42:44 +0200
Subject: [PATCH 09/83] sbus: fix typo

Reviewed-by: Jakub Hrozek <jhrozek@redhat.com>
---
 src/sbus/sbus_message.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/sbus/sbus_message.h b/src/sbus/sbus_message.h
index 99dd930..92d5cea 100644
--- a/src/sbus/sbus_message.h
+++ b/src/sbus/sbus_message.h
@@ -49,7 +49,7 @@ sbus_message_bound(TALLOC_CTX *mem_ctx, DBusMessage *msg);
  *
  * DO NOT USE dbus_message_unref() on such message anymore since it would not
  * release internal data about the bound. The message will be automatically
- * unreferenced whent the talloc context is freed.
+ * unreferenced when the talloc context is freed.
  *
  * @param mem_ctx Memory context to bound the message with. It can not be NULL.
  * @param msg     Message to be bound with memory context.
-- 
2.9.5

