From f12d3379b89bb16ec8e85f496f9dbd9fba95d874 Mon Sep 17 00:00:00 2001
From: Jakub Hrozek <jhrozek@redhat.com>
Date: Mon, 14 May 2012 15:58:37 +0200
Subject: [PATCH 08/19] Use sized_string correctly in FQDN domains

---
 src/responder/nss/nsssrv_cmd.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/responder/nss/nsssrv_cmd.c b/src/responder/nss/nsssrv_cmd.c
index 43e82ae3ad1d98d440c076513ffb78ed46feb949..aa3ef3cbc0b98d3fe44e14dce212ecf1279f14f3 100644
--- a/src/responder/nss/nsssrv_cmd.c
+++ b/src/responder/nss/nsssrv_cmd.c
@@ -1863,7 +1863,7 @@ static int fill_grent(struct sss_packet *packet,
                 if (add_domain) {
                     ret = snprintf((char *)&body[rzero + rsize],
                                     name.len + delim + dom_len,
-                                    namefmt, name, domain);
+                                    namefmt, name.str, domain);
                     if (ret >= (name.len + delim + dom_len)) {
                         /* need more space,
                          * got creative with the print format ? */
@@ -1879,7 +1879,7 @@ static int fill_grent(struct sss_packet *packet,
                         /* retry */
                         ret = snprintf((char *)&body[rzero + rsize],
                                         name.len + delim + dom_len,
-                                        namefmt, name, domain);
+                                        namefmt, name.str, domain);
                     }
 
                     if (ret != name.len + delim + dom_len - 1) {
-- 
1.7.10.1

