From e52a31484c88d46e381238493384f26d9c95f8ff Mon Sep 17 00:00:00 2001
From: Ariel Barria <arielb@fedoraproject.org>
Date: Tue, 22 May 2012 07:13:21 -0500
Subject: [PATCH 15/19] Warn to syslog when dereference requests fail

---
 src/providers/ldap/sdap_async.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/providers/ldap/sdap_async.c b/src/providers/ldap/sdap_async.c
index a8a12c3d390a4ebee0dca81d6610be9fe240a4a6..14a27bcba2385fef5980a5a933cb7e7a9742a231 100644
--- a/src/providers/ldap/sdap_async.c
+++ b/src/providers/ldap/sdap_async.c
@@ -2051,8 +2051,8 @@ static void sdap_deref_search_done(struct tevent_req *subreq)
 
     talloc_zfree(subreq);
     if (ret != EOK) {
-        DEBUG(2, ("dereference processing failed [%d]: %s\n",
-                  ret, strerror(ret)));
+        DEBUG(2, ("dereference processing failed [%d]: %s\n", ret, strerror(ret)));
+        sss_log(SSS_LOG_WARNING, "dereference processing failed : %s", strerror(ret));
         tevent_req_error(req, ret);
         return;
     }
-- 
1.7.10.1

