From c3593f06da54315c88a08a46cfc0def366acad43 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20=C4=8Cech?= <pcech@redhat.com>
Date: Thu, 19 Jan 2017 12:51:27 +0100
Subject: [PATCH 18/79] LDAP: Better logging message

Reviewed-by: Sumit Bose <sbose@redhat.com>
Reviewed-by: Jakub Hrozek <jhrozek@redhat.com>
---
 src/providers/ldap/sdap.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/providers/ldap/sdap.c b/src/providers/ldap/sdap.c
index dc7d5e0caf223c3ee3c43054aa44e796f1b37766..eb460d93bfb067e780868bc9f7bf4e6e0aa1b4a3 100644
--- a/src/providers/ldap/sdap.c
+++ b/src/providers/ldap/sdap.c
@@ -1691,7 +1691,8 @@ static bool sdap_object_in_domain(struct sdap_options *opts,
     sdmatch = sdap_domain_get_by_dn(opts, original_dn);
     if (sdmatch == NULL) {
         DEBUG(SSSDBG_FUNC_DATA,
-              "The group has no original DN, assuming our domain\n");
+              "The original DN of the group cannot "
+              "be related to any search base\n");
         return true;
     }
 
-- 
2.9.3

