From 7c67679ba86682d8c2afea404ec0229641a7f473 Mon Sep 17 00:00:00 2001
From: Lukas Slebodnik <lslebodn@redhat.com>
Date: Mon, 27 Mar 2017 11:59:01 +0200
Subject: [PATCH 40/97] CONFIGURE: Fix fallback if pkg-config for uuid is
 missing
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Reviewed-by: Fabiano Fidêncio <fidencio@redhat.com>
---
 src/external/libuuid.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/external/libuuid.m4 b/src/external/libuuid.m4
index 55411a2118bd787c9d50ba61f9cb791e1c76088d..323521c9224e443f40a15b417038d2dcea9b66f3 100644
--- a/src/external/libuuid.m4
+++ b/src/external/libuuid.m4
@@ -4,7 +4,7 @@ AC_SUBST(UUID_CFLAGS)
 PKG_CHECK_MODULES([UUID], [uuid], [found_uuid=yes], [found_uuid=no])
 
 SSS_AC_EXPAND_LIB_DIR()
-AS_IF([test x"$found_uuid" = xyes],
+AS_IF([test x"$found_uuid" != xyes],
     [AC_CHECK_HEADERS([uuid/uuid.h],
         [AC_CHECK_LIB([uuid],
                       [uuid_generate],
-- 
2.12.2

