From 5a4e7bf91a90e3ece5ef7ad8c2043480902677cc Mon Sep 17 00:00:00 2001
From: Lukas Slebodnik <lslebodn@redhat.com>
Date: Fri, 6 Mar 2015 20:42:35 +0100
Subject: [PATCH 52/99] GPO: Check return value of ad_gpo_store_policy_settings

Reviewed-by: Pavel Reichl <preichl@redhat.com>
---
 src/providers/ad/ad_gpo.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/providers/ad/ad_gpo.c b/src/providers/ad/ad_gpo.c
index 69ff6daaa7e060cca4cab129ecde83d3406702d1..a881741a6ead9244ac123608234d1a0c35f830e3 100644
--- a/src/providers/ad/ad_gpo.c
+++ b/src/providers/ad/ad_gpo.c
@@ -2184,6 +2184,12 @@ ad_gpo_cse_done(struct tevent_req *subreq)
      */
     ret = ad_gpo_store_policy_settings(state->domain,
                                        cse_filtered_gpo->policy_filename);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_OP_FAILURE,
+              "ad_gpo_store_policy_settings failed: [%d](%s)\n",
+              ret, sss_strerror(ret));
+        goto done;
+    }
 
     state->cse_gpo_index++;
     ret = ad_gpo_cse_step(req);
-- 
2.4.0

