From 2d40ce078a9071ac23353848d0849fbbbd600049 Mon Sep 17 00:00:00 2001
From: Jakub Hrozek <jhrozek@redhat.com>
Date: Tue, 30 May 2017 12:19:53 +0200
Subject: [PATCH 85/93] SECRETS: Remove unused declarations
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Reviewed-by: Simo Sorce <simo@redhat.com>
Reviewed-by: Fabiano Fidêncio <fidencio@redhat.com>
Reviewed-by: Lukáš Slebodník <lslebodn@redhat.com>
---
 src/responder/secrets/secsrv.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/responder/secrets/secsrv.h b/src/responder/secrets/secsrv.h
index 3d23c405b3017be83ebc840da5ad253e5269fc7b..0575cbabab07769d2b17f2c0b815004a11b49f90 100644
--- a/src/responder/secrets/secsrv.h
+++ b/src/responder/secrets/secsrv.h
@@ -32,8 +32,6 @@
 
 #define SEC_NET_TIMEOUT 5
 
-struct resctx;
-
 struct sec_ctx {
     struct resolv_ctx *resctx;
     struct resp_ctx *rctx;
-- 
2.14.1

