From dc186bfe90665c13d589b3b4efd9009293e62c46 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pavel=20B=C5=99ezina?= <pbrezina@redhat.com>
Date: Wed, 29 Mar 2017 13:28:49 +0200
Subject: [PATCH 93/97] configure: fix typo
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Reviewed-by: Lukáš Slebodník <lslebodn@redhat.com>
---
 src/external/libhttp_parser.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/external/libhttp_parser.m4 b/src/external/libhttp_parser.m4
index 504bdf0f66c95b3d224c677a205a46e6f8b44726..3a5ef0dbbc63423ad8e960d72e97ec4fb4481dd1 100644
--- a/src/external/libhttp_parser.m4
+++ b/src/external/libhttp_parser.m4
@@ -17,6 +17,6 @@ AS_IF([test x"$found_http_parser" != xyes],
                       ],
                       [-L$sss_extra_libdir -lhttp_parser_strict])],
         [AC_MSG_ERROR([
-You must have the header file http_parse.h installed to build sssd
+You must have the header file http_parser.h installed to build sssd
 with secrets responder. If you want to build sssd without secret responder
 then specify --without-secrets when running configure.])])])
-- 
2.12.2

