From ce68b4ff25cbd52935a540046f0412ce869a27a5 Mon Sep 17 00:00:00 2001
From: Sumit Bose <sbose@redhat.com>
Date: Mon, 9 Oct 2017 09:55:31 +0200
Subject: [PATCH] BUILD: Accept krb5 1.16 for building the PAC plugin
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Reviewed-by: Lukáš Slebodník <lslebodn@redhat.com>
Reviewed-by: Sumit Bose <sbose@redhat.com>
---
 src/external/pac_responder.m4 | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/external/pac_responder.m4 b/src/external/pac_responder.m4
index 6b400d47c919294db51b2e0af767d109f9ba13d7..e0685f0ce33813201b77900b4310a1aa27be121e 100644
--- a/src/external/pac_responder.m4
+++ b/src/external/pac_responder.m4
@@ -17,7 +17,8 @@ then
         Kerberos\ 5\ release\ 1.12* | \
         Kerberos\ 5\ release\ 1.13* | \
         Kerberos\ 5\ release\ 1.14* | \
-        Kerberos\ 5\ release\ 1.15*)
+        Kerberos\ 5\ release\ 1.15* | \
+        Kerberos\ 5\ release\ 1.16*)
             krb5_version_ok=yes
             AC_MSG_RESULT([yes])
             ;;
-- 
2.14.2

