From 7459b3326e99e5d2e6ea924100705ef56525cb96 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fabiano=20Fid=C3=AAncio?= <fidencio@redhat.com>
Date: Mon, 13 Aug 2018 19:34:47 +0200
Subject: [PATCH] sbus_generate: python -> platform-python

---
 sbus_generate.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sbus_generate.sh b/sbus_generate.sh
index 338fd9d33..1eefa4cb5 100755
--- a/sbus_generate.sh
+++ b/sbus_generate.sh
@@ -13,7 +13,7 @@ generate() {
 
     echo "Generating sbus code for: $XML"
 
-    python $CODEGEN --sbus sbus --util util \
+    python3 $CODEGEN --sbus sbus --util util \
         --headers "$HEADERS" \
         --dest "$SRCDIR/src/$DEST" \
         --fileprefix "sbus_${PREFIX}_" \
-- 
2.17.1

