From da891b9cd5a17c65299f84db507181fd74a7a6bf Mon Sep 17 00:00:00 2001
From: Stephen Gallagher <sgallagh@redhat.com>
Date: Thu, 11 Jun 2009 08:46:43 -0400
Subject: [PATCH] Add missing configure check for getpgrp

---
 server/util/signal.m4 |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/server/util/signal.m4 b/server/util/signal.m4
index c6d7f72..a778020 100644
--- a/server/util/signal.m4
+++ b/server/util/signal.m4
@@ -1 +1 @@
-AC_CHECK_FUNCS(sigprocmask sigblock sigaction)
+AC_CHECK_FUNCS(sigprocmask sigblock sigaction getpgrp)
-- 
1.6.2.2

